redis-plugin/src/main/java/org/entando/entando/plugins/jpredis/aps/system/redis/SentinelScheduler.java

```diff
@@ -14,58 +14,69 @@
+import io.lettuce.core.support.caching.CacheFrontend;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
-import java.util.Timer;
 import java.util.TimerTask;
 import org.entando.entando.ent.exception.EntRuntimeException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
 
 /**
  * @author E.Santoboni
  */
 public class SentinelScheduler extends TimerTask {
-    
+
     private static final Logger logger = LoggerFactory.getLogger(SentinelScheduler.class);
-	
-	private Timer timer;
-	private RedisClient lettuceClient;
-	private CacheConfig cacheConfig;
+
+    private final RedisClient redisClient;
+    private final CacheManager cacheManager;
+    private final CacheFrontendManager cacheFrontendManager;
+
     private String currentMasterIp;
-	
-	public SentinelScheduler(RedisClient lettuceClient, int delaySeconds, CacheConfig cacheConfig) {
-        this.cacheConfig = cacheConfig;
-		this.lettuceClient = lettuceClient;
-		this.timer = new Timer();
-        Calendar startTime = Calendar.getInstance();
-        startTime.add(Calendar.SECOND, delaySeconds);
-		this.timer.schedule(this, startTime.getTime(), delaySeconds*1000l);
-        List<Map<String, String>> masters = lettuceClient.connectSentinel().sync().masters();
-        this.currentMasterIp = (!masters.isEmpty()) ? masters.get(0).get("ip") : null;
-        logger.info("CURRENT master node '{}'", this.currentMasterIp);
-	}
-	
-	@Override
-	public boolean cancel() {
-		this.timer.cancel();
-		return super.cancel();
-	}
-	
+
+    public SentinelScheduler(RedisClient redisClient, CacheManager cacheManager,
+            CacheFrontendManager cacheFrontendManager) {
+        this.redisClient = redisClient;
+        this.cacheManager = cacheManager;
+        this.cacheFrontendManager = cacheFrontendManager;
+        try (StatefulRedisSentinelConnection<String, String> connection = redisClient.connectSentinel()) {
+            List<Map<String, String>> masters = connection.sync().masters();
+            this.currentMasterIp = (!masters.isEmpty()) ? masters.get(0).get("ip") : null;
+            logger.info("CURRENT master node '{}'", this.currentMasterIp);
+        }
+    }
+
     @Override
     public void run() {
         try {
-            List<Map<String, String>> masters = lettuceClient.connectSentinel().sync().masters();
-            String ip = (!masters.isEmpty()) ? masters.get(0).get("ip") : null;
-            if (null != this.currentMasterIp && !this.currentMasterIp.equals(ip)) {
-                logger.info("Refresh of front-end-cache -> from master node '{}' to '{}'", this.currentMasterIp, ip);
-                cacheConfig.rebuildCacheFrontend(this.lettuceClient);
-                this.currentMasterIp = ip;
+            try (StatefulRedisSentinelConnection<String, String> connection = redisClient.connectSentinel()) {
```

Comment:
pls insert a log at info level to make to know the operation is executing and maybe also the result ...

Reply:
Added a log that switches from debug to info every 10 invocations, as discussed. Please have  look.